src/libstd/lib.rs

```diff
@@ -211,6 +211,10 @@
 #![deny(intra_doc_link_resolution_failure)]
 #![deny(missing_debug_implementations)]
 
+#![deny(rust_2018_idioms)]
+#![allow(explicit_outlives_requirements)]
+#![allow(elided_lifetimes_in_paths)]
```

Comment:
It is hard work to change all at once (and review it), so `elided_lifetimes_in_paths` is allowed.

I will work on it after this PR is merged.

Reply:
That sounds like a good idea; I think we want to keep `#![allow(explicit_outlives_requirements)]` around for this crate.